### ISSUE TITLE
Add tests for the bot

### ISSUE THREAD

**Issue by bobsilverberg** [mozilla/webqabot#18]
These would likely be unit tests for the internal scripts that the bot needs. This task will involve figuring out how to write these tests, and which tools and/or frameworks to use, as well as writing the tests themselves.

**Comment by MonicaG** [comment]
Note: Hubot project discussing unit testing and updating their documentation about it. https://github.com/github/hubot/issues/985

**Comment by bobsilverberg** [comment]
Interesting - thanks @MonicaG!

**Comment by MonicaG** [comment]
Hi,

I wanted to show you what I've done to with the tests for the qabot to get your feedback.  I've included the code below, but please let me know if you would rather I submit this as a pull request.  

I used the hubot-test-helper and the test uses the mocha testing framework with chai assertions.  I think one outstanding thing would be to make the changes to hook this up to travisci.  I haven't done that before so if you have any tips that would be great.  

Thanks!

```coffeescript
Helper = require('hubot-test-helper')
helper = new Helper('../scripts/webqa.coffee')
chai = require 'chai'
expect = chai.expect

describe 'webqa script', ->    
  hubotVidyoExpectedResponse = [
        [ 'hubot', 'Our Vidyo room is WebQA (8824)']
        [ 'hubot', 'Public link: https://v.mozilla.com/flex.html?roomdirect.html&key=Tc08xVjMQmaVscjhN5jlm7mDknY']
        [ 'hubot', 'Joining by phone: https://wiki.mozilla.org/Teleconferencing#Dialing_In']
        [ 'hubot', 'For information on Vidyo: https://wiki.mozilla.org/Vidyo']
      ]

  beforeEach ->
    @room = helper.createRoom()
    
  afterEach ->
    @room.destroy()

  context 'hubot responding to events', ->
    beforeEach ->
      @room.robotEvent 'announceMeeting', 'in 15 minutes'

    it 'will respond to the announceMeeting event', ->
      expectedMeetingResponse = [
        ['hubot', 'Come join us in 15 minutes for our team meeting!']
        ['hubot', 'Meeting notes are available at https://wiki.mozilla.org/QA/Execution/Web_Testing#Meeting_Notes']
      ]
      expectedMeetingResponse.push hubotVidyoExpectedResponse...
      expect(@room.messages).to.eql expectedMeetingResponse

  context 'hubot responding to user messages', ->
    it 'will respond to meeting', ->
      expectedMeetingResponse = [
        ['shaggy', '@hubot meeting']
        ['hubot', 'Come join us at 9am Pacific every Thursday for our team meeting!']
        ['hubot', 'Meeting notes are available at https://wiki.mozilla.org/QA/Execution/Web_Testing#Meeting_Notes']
      ]
      expectedMeetingResponse.push hubotVidyoExpectedResponse...
      @room.user.say('shaggy', '@hubot meeting').then =>
        expect(@room.messages).to.eql expectedMeetingResponse

    it 'will not respond to meeting when it\'s name is not used', ->
      @room.user.say('shaggy', 'meeting').then =>
        expect(@room.messages).to.eql [
          ['shaggy', 'meeting']
        ]
             
    it 'will respond to list', ->
      @room.user.say('scooby', '@hubot list').then =>
        expect(@room.messages).to.eql [
          ['scooby', '@hubot list']
          ['hubot', 'Email: mozwebqa@mozilla.org']
          ['hubot', 'Subscribe: https://mail.mozilla.org/listinfo/mozwebqa/']
          ['hubot', 'Archive: https://mail.mozilla.org/private/mozwebqa/']
        ]

    it 'will not respond to list when it\'s name is not used', ->
      @room.user.say('scooby', 'list').then =>
        expect(@room.messages).to.eql [
          ['scooby', 'list']
        ]
    
    it 'will respond to mission', ->
      @room.user.say('fred', '@hubot mission').then =>
        expect(@room.messages).to.eql [
          ['fred', '@hubot mission']
          ['hubot', 'Our mission is to provide data, services and tools to positively impact the quality of Mozilla websites. https://wiki.mozilla.org/QA/Execution/Web_Testing/Mission']
        ]

    it 'will not respond to mission when it\'s name is not used', ->
      @room.user.say('fred', 'mission').then =>
        expect(@room.messages).to.eql [
          ['fred', 'mission']
        ]

    it 'will respond to source', ->
      @room.user.say('barney', '@hubot source').then =>
        expect(@room.messages).to.eql [
          ['barney', '@hubot source']
          ['hubot', 'My details and code live at https://github.com/mozilla/webqabot/']
        ]

    it 'will not respond to source when it\'s name is not used', ->
      @room.user.say('barney', 'source').then =>
        expect(@room.messages).to.eql [
          ['barney', 'source']
        ]

    it 'will respond to team', ->
      @room.user.say('wilma', '@hubot team').then =>
        expect(@room.messages).to.eql [
          ['wilma', '@hubot team']
          ['hubot', 'https://wiki.mozilla.org/QA/Execution/Web_Testing#Team_Members']
        ]

     it 'will not respond to team when it\'s name is not used', ->
      @room.user.say('wilma', 'team').then =>
        expect(@room.messages).to.eql [
          ['wilma', 'team']
        ]
    
    it 'will respond to vidyo', ->
      expectedVidyoResult = [['betty', '@hubot vidyo']]
      expectedVidyoResult.push hubotVidyoExpectedResponse...
      @room.user.say('betty', '@hubot vidyo').then =>
        expect(@room.messages).to.eql expectedVidyoResult

    it 'will not respond to vidyo when it\'s name is not used', ->
      @room.user.say('betty', 'vidyo').then =>
        expect(@room.messages).to.eql [
          ['betty', 'vidyo']
        ]
```

**Comment by bobsilverberg** [comment]
Wow, that looks great, @MonicaG! I'd love to try running it myself, so if you could create a PR for it that would be excellent. That's the easiest way for me to grab your code. I think I will have a few questions, but I want to check it out first. As for hooking it up to run on Travis, that's pretty easy. I'm not sure what commands we need to run the actual tests, but Travis config is generally quite straightforward. There are docs available at https://docs.travis-ci.com/, and you can see a couple of examples of our projects that use Travis (albeit for Python rather than JavaScript) at https://github.com/mozilla/oneanddone/blob/master/.travis.yml and https://github.com/mozilla/marketplace-tests/blob/master/.travis.yml.

I'd be happy to answer any specific questions you have about getting it working with Travis, or to even guide you through the process, once you have a PR up.

Thanks again for the excellent work! :+1:

**Comment by MonicaG** [comment]
Hi @bobsilverberg 
Would you please close this issue for me?  I think it didn't close automatically because I added the word "Issue" between "Fixes" and the "#18".  Sorry!

**Comment by bobsilverberg** [comment]
No problem, and thanks again @MonicaG!
